tsl/src/remote/cursor_fetcher.c

```diff
@@ -303,9 +304,22 @@ cursor_fetcher_fetch_data_complete(CursorFetcher *cursor)
 
 		MemoryContextSwitchTo(cursor->state.batch_mctx);
 
+		const char *opt = GetConfigOption("timescaledb.generate_error", true, false);
```

Comment:
Added this config option as a temporary way to inject an error at the right place in the code. Should not be merged with the fix.

To reproduce the crash, first remote the `response = NUL`, then set this session variable:

```
SET timescaledb.generate_error=true;
```

And run an update query against a distributed hypertable.